packages/peer-store/src/utils/address-to-fqmultiaddr.ts

```diff
@@ -0,0 +1,27 @@
+import { CodeError } from '@libp2p/interface/errors'
+import { isMultiaddr } from '@multiformats/multiaddr'
+import { codes } from '../errors.js'
+import type { PeerId } from '@libp2p/interface/peer-id'
+import type { Address } from '@libp2p/interface/peer-store'
+
+/**
+ * Converts addresses for a Peer that may not contain the peerId to addresses
+ * that do contain the peerId.
+ */
+export function addressToFqMultiaddr (peerId: PeerId, addresses: Address[]): Address[] {
+  return addresses.map(({ multiaddr, isCertified }) => {
+    if (!isMultiaddr(multiaddr)) {
+      throw new CodeError('Multiaddr was invalid', codes.ERR_INVALID_PARAMETERS)
+    }
+
+    // is checking for null enough or should we check if peerId.toString() is a mismatch?
+    if (multiaddr.getPeerId() !== peerId.toString()) {
+      multiaddr = multiaddr.encapsulate(`/p2p/${peerId.toString()}`)
+    }
```

Comment:
this was a `== null` check in @achingbrain 's code. I'm wondering if we want to ensure addresses being saved for a particular peer are returning the expected address for that peer.

lmk and I can change back if desired.

Reply:
If there's a peer id present and it's not the peer id we expect maybe we should filter it out?

Otherwise we could end up returning `/ip4/.../tcp/.../p2p/QmFoo/p2p/QmBar` which isn't correct.

Off the top of my head the only time there will be a peer id in the multiaddr that isn't for the peer is circuit relay addresses and since `@multiformats/multaddr@12.1.5` the target peer's address is returned from `getPeerId` rather than the relay.